x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/IndexFollowingIT.java

```diff
@@ -630,6 +630,12 @@ public void testFollowNonExistentIndex() throws Exception {
         );
     }
 
+    public void testFollowWith$InIndexName() throws Exception {
+        String indexSettings = getIndexSettings(1, 0);
+        assertAcked(leaderClient().admin().indices().prepareCreate("test-leader$").setSource(indexSettings, XContentType.JSON).get());
+        followerClient().execute(PutFollowAction.INSTANCE, putFollow("test-leader$", "test-leader$")).actionGet();
```

Comment:
```suggestion
        followerClient().execute(PutFollowAction.INSTANCE, putFollow("test-leader$", "follower-leader$")).actionGet();
```

Should we also check that the follower index actually started?